extract_data.py

```diff
@@ -49,6 +49,17 @@ def extract_and_collect_data_from_generated_files(directory: str, has_dedicated_
         for wifi_card in mobo[1:]:
             wifi_cards.append(wifi_card)
         mobo = mobo[0]
+    def is_a_product(component: dict):
+        #check if brand and model exist
+        if "brand" not in component.keys() or "model" not in component.keys():
+            return False
+        #check if brand or model has a not valid value
+        candidates = [component["brand"].lower(), component["model"].lower()]
+        for candidate in candidates:
+            if isinstance(candidate, str) and (candidate == "" or candidate == "null" or candidate == "unknown" or candidate == "undefined"):
```

Comment:
Where are these strings from?

Reply:
Removing clearly invalid values, I imagine. That's a feature I requested (decode-dimms was reporting Undefined).

To me the chosen values look fine, they clearly cannot be brand names or models and filterint them out should prevent messy situations in case of weird failures from data gathering.